syntax = "proto3";

package tweety;

service SocialGraphService {
  rpc Follow(FollowRequest) returns (FollowResponse);   // RPC to follow a user

  rpc Unfollow(UnfollowRequest) returns (UnfollowResponse);   // RPC to unfollow a user

  rpc GetFollowing(FollowingRequest) returns (FollowingResponse);   // RPC to get a list of users the user is following
  
  rpc GetFollowers(FollowersRequest) returns (FollowersResponse);   // RPC to get a list of followers of a given user
}

message FollowRequest {
  string follower_id = 1;
  string followed_id = 2;
}

message FollowResponse {
  bool success = 1;
  string message = 2;
}

message UnfollowRequest {
  string follower_id = 1;
  string followed_id = 2;
}

message UnfollowResponse {
  bool success = 1;
  string message = 2;
}

message GetFollowingRequest {
  string user_id = 1;
}

message GetFollowingResponse {
  repeated string following_list = 1; // List of usernames the user is following
}

message GetFollowersRequest {
  string user_id = 1;
}

message GetFollowersResponse {
  repeated string followers_list = 1; // List of usernames of followers
}